### DISCUSSION TITLE
SSH Agent

### DISCUSSION THREAD

**Discussion by CoderMcDaniel** [smallstep/certificates#1948] [Q&A]
Wondering what others might be doing to deal with system restarts/reboots in regards to the ssh-agent.   With 'step ssh login', the ssh cert is added to the ssh agent.  However, the ssh-agent and it's state are not persistent through restarts as I understand it.  Is it simply that a new cert will need to be generated and added to the ssh agent or is there another strategy?

Thanks for any help!

**Comment by tashian** [comment]
Yes, the ssh agent does not persist certificates and keys. The feature is, your private keys will only stay in memory.
So, with `step ssh login`, you'll want to run it on-demand.
If you want a longer lived cert for something, you can use `step ssh certificate` to save one to disk.
